cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -619,6 +631,10 @@ func StartControllers(ctx context.Context, controllerCtx ControllerContext, star
 	// - it allows distinguishing between log entries emitted by the controller
 	//   and those emitted for it - this is a bit debatable and could be revised.
 	for controllerName, controllerDesc := range controllerDescriptors {
+		if controllerDesc.RequiresSpecialHandling() {
```

Comment:
I expected to see a similar loop filtering for `RequiresSpecialHandling()` and running those, but there's only one such and it's hard-coded above - why not 

```
for _,  doingSpecialHandling := range []bool{true, false} {
    for controllerName, controllerDesc := range controllerDescriptors {
        if controllerDesc.RequiresSpecialHandling() != doingSpecialHandling {
            continue
        }
    //...
}
```

Reply:
@thockin 

1. I am worried that with this approach, there might be a new controller in the future that needs special handling and would start before ServiceAccountTokenController which we expect to be run first. I thought about adding ordering for the controllers or a graph of dependencies, but that seems like an overkill for now IMO, since we have just one use case.
2. we also have the following line where we initialize cloud provider after the ServiceAccountTokenController is started.
```
	controllerCtx.Cloud.Initialize(controllerCtx.ClientBuilder, ctx.Done())
```